src/librustdoc/clean/mod.rs

```diff
@@ -1730,9 +1737,7 @@ impl Clean<Variant> for hir::VariantData<'_> {
     fn clean(&self, cx: &mut DocContext<'_>) -> Variant {
         match self {
             hir::VariantData::Struct(..) => Variant::Struct(self.clean(cx)),
-            hir::VariantData::Tuple(..) => {
-                Variant::Tuple(self.fields().iter().map(|x| x.ty.clean(cx)).collect())
-            }
+            hir::VariantData::Tuple(..) => Variant::Struct(self.clean(cx)),
```

Comment:
Why this change? Is `Tuple` ever used now? If not, can it be removed?

Reply:
It is:

```
failures:
    [rustdoc] rustdoc/const-generics/const-generics-docs.rs
    [rustdoc] rustdoc/intra-doc/basic.rs
    [rustdoc] rustdoc/issue-85454.rs
    [rustdoc] rustdoc/synthetic_auto/overflow.rs
    [rustdoc] rustdoc/toggle-item-contents.rs
    [rustdoc] rustdoc/tuple-struct-fields-doc.rs
    [rustdoc] rustdoc/type-layout.rs
    [rustdoc] rustdoc/viewpath-rename.rs
    [rustdoc] rustdoc/viewpath-self.rs
    [rustdoc] rustdoc/where.rs
```

Leaving it then.

Reply:
Hmm. Where is it used, and why did you change this code from `Tuple` to `Struct`?

Reply:
So, `Variant::Tuple` is used on tuple structs which are not in an enum. In case we are in an enum, the kind is provided by the `Variant` wrapper directly, and since we need the fields' name (yes, even for a tuple struct variant), I simply switched to that to make things simpler.

Do you want me to use `Variant::Tuple` anyway (meaning the code will require refactorings in multiple places just like I originally did) or do you prefer to keep things as is (and then I add documentation to explain this)? The first solution requires a lot of changes but nothing impossible (I already did it and then reverted it afterwards for this solution) but might improve code clarity. So as you prefer.

Reply:
Nah, this is fine - can you add a comment?

Reply:
Sure!